### PR TITLE
Addition of the reader of the NYPL

### PR DESCRIPTION
**Pull request by Scarsniik**

### CODE REVIEW

File: reader-NYPL/webapp.webmanifest [hunk 1]
```diff
@@ -0,0 +1 @@
+{"short_name":"Ulysses","name":"Ulysses","start_url":"index.html","display":"standalone","icons":[{"src":"images/cover.png","size":"144x144","type":"image/png"}]}
```
Comment (danielweck) [inline reader-NYPL/webapp.webmanifest]: I suggest removing / disabling this file (unnecessary in Electron)

File: reader-NYPL/sw.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+var CACHE_NAME = "webpub-viewer";
```
Comment (danielweck) [inline reader-NYPL/sw.js]: I suggest removing / disabling Service Worker (caching functionality) in Electron

File: reader-NYPL/manifest.appcache [hunk 3]
```diff
@@ -0,0 +1,14 @@
+CACHE MANIFEST
```
Comment (danielweck) [inline reader-NYPL/manifest.appcache]: I suggest removing / disabling AppCache in Electron app

File: reader-NYPL/appcache.html [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<html manifest="manifest.appcache">
```
Comment (danielweck) [inline reader-NYPL/appcache.html]: I suggest removing / disabling App Cache in Electron

File: reader-NYPL/index.html [hunk 5]
```diff
@@ -0,0 +1,90 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <title>Webpub manifest viewer</title>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
+  <meta name="apple-mobile-web-app-capable" content="yes">
+  <link href="webapp.webmanifest" rel="manifest" type="application/manifest+json">
```
Comment (danielweck) [inline reader-NYPL/index.html]: I suggest removal of Web App Manifest in this Electron app

File: reader-NYPL/index.html [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+      return params;
+    };
+
+    require(["LocalStorageStore", "ServiceWorkerCacher", "IFrameNavigator", "ColumnsPaginatedBookView", "ScrollingBookView", "LocalAnnotator", "BookSettings"],
+            function (LocalStorageStore, ServiceWorkerCacher, IFrameNavigator, ColumnsPaginatedBookView, ScrollingBookView, LocalAnnotator, BookSettings) {
+      var element = document.getElementById("viewer");
+      var urlParams = getURLQueryParams();
+      var webpubManifestUrl = new URL(urlParams['url']);
+      var store = new LocalStorageStore.default({ prefix: webpubManifestUrl.href });
+      var cacher = new ServiceWorkerCacher.default({
```
Comment (danielweck) [inline reader-NYPL/index.html]: I suggest disabling / removing Service Worker caching in this Electron app

File: reader-NYPL/index.html [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        store: settingsStore,
+        bookViews: [paginator, scroller],
+        fontSizesInPixels: fontSizes,
+        defaultFontSizeInPixels: defaultFontSize
+      }).then(function (settings) {
+        IFrameNavigator.default.create({
+          element: element,
+          manifestUrl: webpubManifestUrl,
+          store: store,
+          cacher: cacher,
```
Comment (danielweck) [inline reader-NYPL/index.html]: Once Service Worker caching is remove, this can be set to `null`

File: src/components/App.tsx [hunk 8]
```diff
@@ -53,11 +60,24 @@ export default class App extends React.Component<undefined, AppState> {
     }
 
     public render(): React.ReactElement<{}> {
+        let manifestUrl = "https://readium2.herokuapp.com/pub/L2FwcC9jaGlsZHJlbnMtbGl0ZXJhdHVyZS5lcHVi/manifest.json";
         return (
```
Comment (danielweck) [inline src/components/App.tsx]: Just out of interest: have you managed to integrate `r2-streamer-js` in the Electron app (as a NPM package). Depending on which Electron version you are using at the moment (see latest https://github.com/electron/electron/blob/master/.node-version ) you will be able to pick a particular build target above ES6-2015, see https://unpkg.com/r2-streamer-js/dist/

File: reader-NYPL/index.html (hunk 6 above)
Comment (Scarsniik) [inline reader-NYPL/index.html]: I don't understand how to remove this

Comment (danielweck) [inline reader-NYPL/index.html]: I haven't tried myself, but I imagine passing `null` instead of `cacher` should do?

Comment (Scarsniik) [inline reader-NYPL/index.html]: If I pass null theire is an error

Comment (danielweck) [inline reader-NYPL/index.html]: Ah, that's annoying. It sounds like we eventually should fork NYPL's `webpub-viewer` https://github.com/NYPL-Simplified/webpub-viewer in order to customise it for our particular use in Electron (in a nutshell: removal of all the web-centric features such as web app manifest, as well as HTTP caching, which we don't really need).

**Pull request merged**